ambry-api/src/main/java/com.github.ambry/account/Container.java

```diff
@@ -0,0 +1,226 @@
+    }
+    if (parentAccount == null) {
+      throw new IllegalArgumentException("parentAccount cannot be null.");
+    }
+    this.id = id;
+    this.name = name;
+    this.status = status;
+    this.description = description;
+    this.isPrivate = isPrivate;
+    this.parentAccount = parentAccount;
+    metadata = new JSONObject();
+    metadata.put(CONTAINER_METADATA_VERSION_KEY, CONTAINER_METADATA_VERSION_1);
+    metadata.put(CONTAINER_ID_KEY, id);
+    metadata.put(CONTAINER_NAME_KEY, name);
+    metadata.put(CONTAINER_STATUS_KEY, status);
+    metadata.put(CONTAINER_DESCRIPTION_KEY, description);
+    metadata.put(CONTAINER_IS_PRIVATE_KEY, isPrivate);
+  }
+
+  /**
+   * Gets the id of the container.
+   * @return The id of the container.
+   */
+  public short getId() {
+    return id;
+  }
+
+  /**
+   * Gets the name of the container.
+   * @return The name of the container.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Gets the status of the container.
+   * @return The status of the container.
+   */
+  public String getStatus() {
+    return status;
+  }
+
+  /**
+   * Gets the description of the container.
+   * @return The description of the container.
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the boolean value if the entries in this container are private.
+   * @return A boolean indicating if the entries in this container are private.
+   */
+  public boolean getPrivate() {
+    return isPrivate;
+  }
+
+  /**
+   * Gets the {@link Account} that owns this container.
+   * @return The parent {@link Account} of this container.
+   */
+  public Account getParentAccount() {
+    return parentAccount;
+  }
+
+  /**
+   * Gets the metadata of the container.
+   * @return The metadata of the container.
+   */
+  public JSONObject getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Generates a {@link String} representation that uniquely identifies this container. The string
+   * is in the format of {@code Container[accountId:accountName:containerId:containerName]}.
+   * @return The {@link String} representation of this container.
+   */
+  @Override
+  public String toString() {
+    return "Container[" + parentAccount.getId() + ":" + parentAccount.getName() + ":" + getId() + ":" + getName() + "]";
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Container)) {
+      return false;
+    }
+
+    Container container = (Container) o;
+
+    if (id != container.id) {
+      return false;
+    }
+    if (isPrivate != container.isPrivate) {
+      return false;
+    }
+    if (!name.equals(container.name)) {
+      return false;
+    }
+    if (status != null ? !status.equals(container.status) : container.status != null) {
+      return false;
+    }
+    if (description != null ? !description.equals(container.description) : container.description != null) {
```

Comment:
I think so, it will be odd if two containers considered to be equal only if their descriptions are not the same.